instrumentation/mongo/mongo-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/mongo/MongoClientTracer.java

```diff
@@ -154,6 +154,16 @@ private JsonWriterSettings createJsonWriterSettings(int maxNormalizedQueryLength
       }
     } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException ignored) {
     }
+    if (settings == null) {
+      try {
+        settings = JsonWriterSettings.class.getConstructor(Boolean.TYPE).newInstance(false);
+      } catch (InstantiationException
+          | IllegalAccessException
+          | InvocationTargetException
+          | NoSuchMethodException ignored) {
+      }
```

Comment:
What happens if this is null, is it ok?

Reply:
no, null is not ok, we will get NPE later. But null shouldn't happen because constructor with boolean exists prior version 4.0. And since 4.0 there is a builder only.